Ical.Net/DataTypes/CalDateTime.cs

```diff
@@ -510,67 +487,25 @@ public IDateTime AddDays(int days)
     }
 
     /// <inheritdoc cref="DateTime.AddHours"/>
-    /// <remarks>
-    /// This will also add a <see cref="IDateTime.Time"/> part that did not exist before the operation,
-    /// if the hours are not a multiple of 24.
-    /// </remarks>
-    public IDateTime AddHours(int hours)
-    {
-        var copy = Copy<CalDateTime>();
-        var newValue = copy.Value.AddHours(hours);
-        if (!copy.HasTime && (hours % 24 == 0))
-        {
-            copy._dateOnly = DateOnly.FromDateTime(newValue);
-        }
-        else
-        {
-            copy._dateOnly = DateOnly.FromDateTime(newValue);
-            copy._timeOnly = TruncateTimeToSeconds(newValue);
-        }
-        return copy;
-    }
+    /// <exception cref="InvalidOperationException">
+    /// Thrown when attempting to add a time span to a date-only instance, 
+    /// and the time span is not a multiple of full days.
+    /// </exception>
+    public IDateTime AddHours(int hours) => Add(TimeSpan.FromHours(hours));
```

Comment:
Do we need the individual Add... variants? Maybe just remove and use the `Add(TimeSpan)` instead?

Reply:
tracked with #677